### PR TITLE
[Tune] Remove Ray Client references from Tune and Train docs/examples

### LINKED ISSUE #30227: Issue on page /tune/examples/pbt_ppo_example.html
The output of the example has errors and also how to specify the scaling parameters like num_cpus, num_gpus and resources_per_trial here

### LINKED ISSUE #31621: [client][ml] Remove Ray Client references from Libraries Documentation and Examples
See https://github.com/ray-project/ray/issues/31516 for details.

This ticket tracks the work needed to remove Ray Client references specific to Train and Tune.

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

This PR removes references to Ray Client in Tune and Train examples. It also removes outdated references of needing `ray.init("auto")` being used to connect to an existing cluster vs. `ray.init()` creating a new local cluster.

The latest `ray.init()` docstring explains that:

> This method handles two cases; either a Ray cluster already exists and we just attach this driver to it or we start all of the processes associated with a Ray cluster and attach to the newly started cluster.

New version of this PR: https://github.com/ray-project/ray/pull/31712

## Related issue number

Closes https://github.com/ray-project/ray/issues/31621
Closes https://github.com/ray-project/ray/issues/30227 (fixes the `pbt_ppo_example.ipynb` config to use a valid Humanoid env, and cleans up the display output)

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/tune/examples/pbt_transformers.ipynb [hunk 1]
```diff
@@ -184,38 +184,20 @@
...
     "    parser = argparse.ArgumentParser()\n",
     "    parser.add_argument(\n",
-    "        \"--smoke-test\", default=True, action=\"store_true\", help=\"Finish quickly for testing\"\n",
+    "        \"--smoke-test\",\n",
+    "        default=True,\n",
+    "        action=\"store_true\",\n",
+    "        help=\"Finish quickly for testing\",\n",
     "    )\n",
-    "    parser.add_argument(\n",
-    "        \"--ray-address\",\n",
```
Comment (richardliaw) [inline doc/source/tune/examples/pbt_transformers.ipynb:190]: i thought we were going to keep this per previous comments?

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
Are tests failures related?

**Comment by justinvyu** [comment]
@richardliaw Yep, those are rllib tests that aren't touched by this PR.

**Review by krfricke — APPROVED** [review]
Thanks!

**Pull request merged**